src/sentry/static/sentry/app/views/settings/organizationRelays/modals/modalManager.tsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import omit from 'lodash/omit';
+import isEqual from 'lodash/isEqual';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import {ModalRenderProps} from 'app/actionCreators/modal';
+import {Organization, Relay} from 'app/types';
+
+import Form from './form';
+import Modal from './modal';
+import handleXhrErrorResponse from './handleXhrErrorResponse';
+
+type FormProps = React.ComponentProps<typeof Form>;
+type Values = FormProps['values'];
+
+type Props = ModalRenderProps & {
+  onSubmitSuccess: (organization: Organization) => void;
+  orgSlug: Organization['slug'];
+  api: Client;
+  savedRelays: Array<Relay>;
+};
+
+type State = {
+  values: Values;
+  requiredValues: Array<keyof Values>;
+  disables: FormProps['disables'];
+  errors: FormProps['errors'];
+  isFormValid: boolean;
+  title: string;
+};
+
+class DialogManager<
+  P extends Props = Props,
+  S extends State = State
+> extends React.Component<P, S> {
+  constructor(props: P) {
+    super(props);
+    this.handleValidate = this.handleValidate.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleValidateKey = this.handleValidateKey.bind(this);
+    this.handleSave = this.handleSave.bind(this);
```

Comment:
Should we replace these with an arrow functions?

Reply:
I'd love too, but I got a feedback from Evan saying that arrow funcs inside classes are much slower than bound funcs ...because of that I decided to convert them

Reply:
Do you mean this comment? https://github.com/getsentry/sentry/pull/19506#discussion_r444437158

Reply:
+1 to what @matejminar said, in that case there was no reason to bind `this`, it's unrelated to performance